compiler/optimizer/OMRCFGSimplifier.cpp

```diff
@@ -169,6 +156,594 @@ bool TR_CFGSimplifier::simplifyBooleanStore()
+   comp()->getCheckcastNullChkInfo().push_front(bcInfo);
+   checkcastAndNULLCHKNode->setAndIncChild(0, objNode);
+   checkcastAndNULLCHKNode->setAndIncChild(1, classNode);
+
+   if (trace())
+      traceMsg(comp(), "Remove compareTreeTop n%dn\n", compareTreeTop->getNode()->getGlobalIndex()); 
+   TR::TransformUtil::removeTree(comp(), compareTreeTop);
+
+   TR::TreeTop *checkcastAndNULLCHKTree = TR::TreeTop::create(comp(), checkcastAndNULLCHKNode);
+
+   if (trace())
+      traceMsg(comp(), "Create checkcastAndNULLCHK Node n%dn\n", checkcastAndNULLCHKNode->getGlobalIndex()); 
+
+   _block->append(checkcastAndNULLCHKTree);
+
+   if (hasExceptionPoint(_block, checkcastAndNULLCHKTree))
+      _block = _block->split(checkcastAndNULLCHKTree, _cfg, true, false);
+
+   _cfg->addNode(catchBlock);
+   _cfg->addExceptionEdge(_block, catchBlock);
+   _cfg->addEdge(catchBlock, throwBlock);
+   _cfg->removeEdge(_block, throwBlock);
+   
+   if (_block->getNextBlock() != fallthroughBlock) {
+      TR::Node *gotoNode = TR::Node::create(checkcastAndNULLCHKNode, TR::Goto, 0);
+      gotoNode->setBranchDestination(fallthroughBlock->getEntry());
+      _block->append(TR::TreeTop::create(comp(), gotoNode));
+   }
+
+   if (trace())
+      traceMsg(comp(), "End simplifyInstanceOfTestToCheckcast.\n");
+
+   TR::DebugCounter::incStaticDebugCounter(comp(), TR::DebugCounter::debugCounterName(comp(), "cfgSimpCheckcast/(%s)", comp()->signature()));
+   
+   return true;
+   }
+
+static bool containsIndirectOperationImpl(TR::Node *node, TR::NodeChecklist *visited, int32_t depth)
+   {
+   if (visited->contains(node))
+      return false;
+
+   if (depth == 0)
+      return true;
+
+   visited->add(node);
+
+   if (!(node->getOpCode().isArithmetic() && !node->getOpCode().isDiv())
+       && !node->getOpCode().isLoadVarDirect()
+       && !node->getOpCode().isTernary()
+       && !node->getOpCode().isLoadConst())
+      return true;
+
+   if (node->getOpCode().hasSymbolReference()
+       && !node->getSymbolReference()->getSymbol()->isAutoOrParm())
+      return true;
+
+   /*if (node->getOpCode().isIndirect()
+       || node->getOpCodeValue() == TR::arraylength)
+      return true;
+
+   if (node->getOpCode().hasSymbolReference()
+       && (node->getSymbol()->isVolatile()
+           || !node->getOpCode().isLoadDirect()))
+      return true;*/
+
+   for (int i = 0; i < node->getNumChildren(); ++i)
+      {
+      if (containsIndirectOperationImpl(node->getChild(i), visited, depth-1))
+         return true;
+      }
+   return false;
+   }
+
+static bool containsIndirectOperation(TR::Compilation *comp, TR::TreeTop *treetop)
+   {
+   TR::NodeChecklist visited(comp);
+   return containsIndirectOperationImpl(treetop->getNode()->getFirstChild(), &visited, 3);
+   }
+
+bool OMR::CFGSimplifier::simplifyCondStoreSequence(bool needToDuplicateTree)
+   {
+   static char *disableSimplifyCondStoreSequence = feGetEnv("TR_disableSimplifyCondStoreSequence");
```

Comment:
Is there a reason for using an environment variable instead of an option here?

Reply:
Commented previously.